airflow/www/views.py

```diff
@@ -844,7 +844,7 @@ def index(self):
 
         with create_session() as session:
             # read orm_dags from the db
-            dags_query = select(DagModel).where(~DagModel.is_subdag, DagModel.is_active)
+            dags_query = select(DagModel).where(DagModel.is_active)
```

Comment:
This combined with the model change means all previously subDAGs will not show up in the index page as normal DAGs. Should we provide a way for users to remove all those subDAGs?

Reply:
We could but that would require this PR to not remove the db field (just yet), otherwise it would be impossible to tell what DAGs should be removed.

Reply:
The Migration Utilities could be run before upgrading to 3.x -- so the environment would be ready for 3.0. 

<img width="557" alt="image" src="https://github.com/user-attachments/assets/56c6bc6c-5ba7-460d-b0fc-fbe58ff15a3b">

Reply:
(Similar to how we had it for 2.0 -- Step 3 in the above figure is where we ran it whereas we upgrade on Step 7)

https://airflow.apache.org/docs/apache-airflow/2.0.0/upgrading-to-2.html

Reply:
I could also delete those records in the DB migration if we'd like that

Reply:
Unconditionally deleting all subDAGs is a bit too heavy-handed to me. We could do that _if_ the migration tool allows the user to mark what DAGs to not be removed (maybe by setting their `is_subdag` flag to False somehow?)

Reply:
In that case, for now, we should rely on the migration tool to delete or change it to main DAG IMO.